Lib/test/test_ntpath.py

```diff
@@ -637,6 +638,36 @@ def test_realpath_cwd(self):
         with os_helper.change_cwd(test_dir_short):
             self.assertPathEqual(test_file_long, ntpath.realpath("file.txt"))
 
+    @unittest.skipUnless(HAVE_GETFINALPATHNAME, 'need _getfinalpathname')
+    def test_realpath_permission(self):
+        # Test whether python can resolve the real filename of a
+        # shortened file name even if it does not have permission to access it.
+        ABSTFN = ntpath.abspath(os_helper.TESTFN)
+
+        os_helper.unlink(ABSTFN)
+        os_helper.rmtree(ABSTFN)
+        os.mkdir(ABSTFN)
+        self.addCleanup(os_helper.rmtree, ABSTFN)
+
+        test_file = ntpath.join(ABSTFN, "LongFileName123.txt")
+        test_file_short = ntpath.join(ABSTFN, "LONGFI~1.TXT")
+
+        with open(test_file, "wb") as f:
+            f.write(b"content")
```

Comment:
Automatic generation of short names may be disabled on NTFS volumes for the sake of performance. (They're not supported at all on ReFS and exFAT.) Try to set the short name using the following command line:

```python
subprocess.run(['fsutil.exe', 'file', 'setShortName', test_file, 'LONGFI~1.TXT'], creationflags=subprocess.DETACHED_PROCESS)
```

Reply:
You're right! Do you think I should check the drive type with `fsutil fsinfo volumeinfo X:`?

Reply:
Try to run the above command to set the short name, without worry about whether it succeeds. Then if `ntpath.realpath(test_file_short))` isn't the same as `test_file`, skip the test with the message that the filesystem lacks the required support for short filenames.

Reply:
Ah, thus `run()` instead of `check_call()`. OK, wilco.